Github Authenticator

5290charlie opened the pull request:
Starting point for github authenticator

Update naming and parse response to symbolized hash

Return gh access_token

Update gh token reference

5290charlie commented on app/controllers/oauth2_controller.rb:
```diff
@@ -76,6 +78,18 @@ def authenticate_with_edx(username, auth_code)
+      auth = GithubAuthenticator.new(auth_state, auth_code)
+      login = auth.authenticate!
+
+      render json: { access_token: login.oauth2_token, github_token: auth.github_token }
```
Wanted ability to utilize the user's actual github oauth token for queries against github's api on their behalf ... maybe a better way to achieve this?

marcoow replied:
I believe the OAuth 2.0 spec doesn’t allow to respond with any custom attributes. After all, github would just be used as an identity provider here anyway. That doesn’t mean github‘s ath Code couldn’t be persisted and exposed to the client but I think this is not the right endpoint and it’s probably out of scope for rails_api_auth.

marcoow commented on app/services/github_authenticator.rb:
```diff
@@ -0,0 +1,68 @@
+        provider: PROVIDER
+      }
+
+      Login.create!(login_attributes)
```
I think once we ever get around to extract the ˋIdentityˋ model from the ˋLoginˋ model (see #15), that new model would be the ideal place to persist the github auth token so it could be used (and potentially be exposed to the client) later.

5290charlie commented:
@marcoow I have updated this PR to remove the `github_token` from the auth response, and am excited to see the identities work is in progress!

I also updated the tests to fit the new GitHub authenticator.

5290charlie commented:
specs are now passing, but rubocop is blowing up on the  ... I'll let you guys update any rubocop config or syntax changes 👍

marcoow commented:
Yay 🎉 

I created a PR to fix the Rubocop offenses: #68. Once that's merged this can be rebased and should be good to go.
